File: ProiectCS/ProiectCS/services/TicketServices.cs
using System.Collections;
using System.Collections.Generic;
using NET.model;
using NET.model.validators;
using NET.repository;

namespace ProiectCS.services
{
    public class TicketServices
    {
        private TicketRepository ticketRepository;
        private Validator<Ticket> ticketValidator;
        private FlightServices flightServices;

        public TicketServices(TicketRepository ticketRepository, Validator<Ticket> ticketValidator, FlightServices flightServices)
        {
            this.ticketRepository = ticketRepository;
            this.ticketValidator = ticketValidator;
            this.flightServices = flightServices;
        }

        public void addTicket(string customerName, string touristName, string customerAddress, int seats, Flight flight)
        {
            Flight flightForTicket = flightServices.findFlightByID(flight.ID);
            Ticket saveTicket = new Ticket(customerName, touristName, customerAddress, seats, flightForTicket);
            ticketValidator.validate(saveTicket);
            
            flightServices.updateFlight(flightForTicket.Destination, flightForTicket.Departure, 
                flightForTicket.Airport,flightForTicket.AvailableSeats - seats,flightForTicket.ID);
            ticketRepository.Save(saveTicket);
        }

        public void deleteTicket(string customerName, string touristName, string customerAddress, 
            int seats, Flight flight, int Id)
        {
            Ticket removeTicket = new Ticket(Id, customerName, touristName, customerAddress, seats, flight);
            ticketRepository.Delete(removeTicket);
        }

        public void updateTicket(string customerName, string touristName, string customerAddress,
            int seats, Flight flight, int Id)
        {
            Ticket upTicket = new Ticket(Id,customerName,touristName,customerAddress,seats,flight);
            ticketValidator.validate(upTicket);
            ticketRepository.Update(upTicket, Id);
        }

        public Ticket findTicketById(int Id)
        {
            return ticketRepository.FindById(Id);
        }

        public IEnumerable<Ticket> findAllTickets()
        {
            return ticketRepository.FindAll();
        } 
        
    }
}

File: Java/src/main/java/services/TicketServices.java
package services;

import model.Flight;
import model.Ticket;
import model.validators.TicketValidator;
import repository.TicketRepository;

import java.util.Collection;

public class TicketServices {
    private TicketRepository ticketRepository;
    private TicketValidator ticketValidator;
    private FlightServices flightServices;

    public TicketServices(TicketRepository ticketRepository,TicketValidator ticketValidator
            ,FlightServices flightServices) {
        this.ticketRepository = ticketRepository;
        this.ticketValidator = ticketValidator;
        this.flightServices = flightServices;
    }

    public void addTicket(String customerName, String touristName, String customerAddress, Integer seats, Flight flight){
        Flight flightForTicket = flightServices.findFlightByID(flight.getID());
        Ticket saveTicket = new Ticket(customerName,touristName,customerAddress,seats,flightForTicket);
        ticketValidator.validate(saveTicket);

        flightServices.updateFlight(flight.getDestination(), flight.getDeparture(),
                flight.getAirport(), flight.getAvailableSeats() - seats, flight.getID());
        ticketRepository.add(saveTicket);
    }

    public void deleteTicket(String customerName, String touristName, String customerAddress, Integer seats, Flight flight, Integer Id){
        Ticket removeTicket = new Ticket(Id,customerName,touristName,customerAddress,seats,flight);
        ticketRepository.delete(removeTicket);
    }

    public void updateTicket(String customerName, String touristName, String customerAddress, Integer seats, Flight flight, Integer Id){
        Ticket upTicket = new Ticket(Id,customerName,touristName,customerAddress,seats,flight);
        ticketValidator.validate(upTicket);
        ticketRepository.update(upTicket,Id);
    }

    public Ticket findTicketByID(Integer Id){
        return ticketRepository.findByID(Id);
    }

    public Iterable<Ticket> findAllTickets(){
        return ticketRepository.findAll();
    }

    public Collection<Ticket> getAllTickets(){
        return ticketRepository.getAll();
    }
}
